tests/system/chatbot/test_chatbot.py

```diff
@@ -1,48 +1,70 @@
+import os
+
 import pytest
-import requests
-from xprocess import ProcessStarter
 
+from jina import Flow, Document
+from jina.types.document.generators import from_csv
+from jina.helloworld.chatbot.app import hello_world
 from jina.parsers.helloworld import set_hw_chatbot_parser
+from jina.helloworld.chatbot.app import download_data
+from jina.helloworld.chatbot.my_executors import MyTransformer, MyIndexer
+from tests import validate_callback
 
 
 @pytest.fixture
-def chatbot_args(tmpdir):
+def helloworld_args(tmpdir):
     return set_hw_chatbot_parser().parse_args(
-        ['--workdir', str(tmpdir), '--port-expose', '8080']
+        [
+            '--workdir',
+            str(tmpdir),
+            '--port-expose',
+            '8080',
+            '--unblock-query-flow',
+            '--parallel',
+            '1',
+        ]
     )
 
 
-@pytest.fixture
-def payload():
-    return {'top_k': 1, 'data': ['text:Is my dog safe from virus']}
-
-
-@pytest.fixture
-def post_uri():
-    return 'http://localhost:8080/search'
-
-
-@pytest.fixture
-def expected_result():
-    return '''It's not completely up to you.'''
+def test_chatbot(helloworld_args, tmpdir, mocker):
+    """Regression test for chatbot example."""
+    hello_world(helloworld_args)
 
+    def validate_response(resp):
+        assert len(resp.data.docs) == 1
+        for doc in resp.data.docs:
+            assert len(doc.matches) == 1
+            assert (
+                'there is currently no evidence that pets such as dogs or cats can be'
+                in doc.matches[0].tags['answer']
+            )
 
-@pytest.fixture(autouse=True)
-def start_server(xprocess, chatbot_args):
-    class Starter(ProcessStarter):
-        pattern = "You should see a demo page opened in your browser"
-        args = ["jina", "hello", "chatbot"]
-        max_read_lines = 10000
+    targets = {
+        'covid-csv': {
+            'url': helloworld_args.index_data_url,
+            'filename': os.path.join(helloworld_args.workdir, 'dataset.csv'),
+        }
+    }
 
-    xprocess.ensure("server", Starter)
-    yield
-    xprocess.getinfo("server").terminate()
+    mock_on_done = mocker.Mock()
+    mock_on_fail = mocker.Mock()
 
+    download_data(
```

Comment:
I think we do not need to `download` again. We can `index` some set of `text` and `query` with something, not necessarily the same data as in the `example`? It would take too long otherwise